packages/angular/projects/angular-sdk/README.md

```diff
@@ -0,0 +1,206 @@
+```
+
+#### Required peer dependencies
+
+The following list contains the peer dependencies of `@openfeature/angular-sdk`.
+See the [package.json](./package.json) for the required versions.
+
+* `@openfeature/web-sdk`
+* `@angular/common`
+* `@angular/core`
+
+#### Usage
+
+#### Module
+
+To include the OpenFeature Angular directives in your application, you need to import the `OpenFeatureModule` and configure it using the `forRoot` method.
+
+```typescript
+import { NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { OpenFeatureModule } from '@openfeature/angular-sdk';
+
+@NgModule({
+  declarations: [
+    // Other components
+  ],
+  imports: [
+    CommonModule,
+    OpenFeatureModule.forRoot({
+      provider: yourFeatureProvider,
+      domainBoundProviders: {
+        domain1: new YourOpenFeatureProvider(),
+        domain2: new YourOtherOpenFeatureProvider(),
+      },
+    })
+  ],
+})
+export class AppModule {
+}
+```
+
+##### Minimal Example
+
+You don't need to provide all the templates. Here's a minimal example using a boolean feature flag:
+
+If `initializing` and `reconciling` are not given, the feature flag value that is returned by the provider will determine what will be rendered.
+
+```html
+<div *booleanFeatureFlag="'isFeatureEnabled'; default: true">
+  This is shown when the feature flag is enabled.
+</div>
+```
+
+This example shows content when the feature flag `isFeatureEnabled` is true with a default value of true. No `else`, `initializing`, or `reconciling` templates are required in this case.
+
+#### Flag Types
+
+The library provides four primary directives for feature flags:
+
+##### Boolean Feature Flag
+
+```html
+<div *booleanFeatureFlag="'isFeatureEnabled'; default: true; domain: 'userDomain'; else: booleanFeatureElse; initializing: booleanFeatureInitializing; reconciling: booleanFeatureReconciling">
+  This is shown when the feature flag is enabled.
+</div>
+<ng-template #booleanFeatureElse>
+  This is shown when the feature flag is disabled.
+</ng-template>
+<ng-template #booleanFeatureInitializing>
+  This is shown when the feature flag is initializing.
+</ng-template>
+<ng-template #booleanFeatureReconciling>
+  This is shown when the feature flag is reconciling.
+</ng-template>
+```
+
+##### Number Feature Flag
+
+```html
+<div *numberFeatureFlag="'discountRate'; value: 10; default: 5; domain: 'userDomain'; else: numberFeatureElse; initializing: numberFeatureInitializing; reconciling: numberFeatureReconciling">
+  This is shown when the feature flag matches the specified discount rate.
+</div>
+<ng-template #numberFeatureElse>
+  This is shown when the feature flag does not match the specified discount rate.
+</ng-template>
+<ng-template #numberFeatureInitializing>
+  This is shown when the feature flag is initializing.
+</ng-template>
+<ng-template #numberFeatureReconciling>
+  This is shown when the feature flag is reconciling.
+</ng-template>
+```
```

Comment:
Something that I think could be nice here, is to add an `exportAs` value for this directive (`exportAs: 'evaluationDetails'`), and bind the flag details to that... then in the templates, you could access and render the flag values like `#details="evaluationDetails"` and then `{{ details.value }}`

I'm not sure if this would be valuable for booleans, but I think it would be for others.

Reply:
Thinking about it more, I think that the first 2 templates don't make so much sense for non-booleans. I think most users won't actually be interested in showing a template if the string/number/object value matches. I think it would be more useful just to have 3 templates for these: a primary(with flag value injected)/initializing/reconciling template.

I'm not also not 100% sure there's much value in differentiating between the Initializing and reconciling templates; in react, we just show the suspense in both cases and I suspect that's probably all people would need.

Reply:
> Something that I think could be nice here, is to add an exportAs value for this directive (exportAs: 'evaluationDetails'), and bind the flag details to that... then in the templates, you could access and render the flag values like #details="evaluationDetails" and then {{ details.value }}

Definitely, but it was already in. I added some docs for that :) 
https://github.com/open-feature/js-sdk/pull/997/files#diff-41afcfae7e1b2d20dffab3b5bd15a8858a79acb6f7b468e45f154e9eecf2d539R231

Reply:
> Thinking about it more, I think that the first 2 templates don't make so much sense for non-booleans. I think most users won't actually be interested in showing a template if the string/number/object value matches. I think it would be more useful just to have 3 templates for these: a primary(with flag value injected)/initializing/reconciling template.

I see the point. We actually have the use case of conditionally rendering based on the value with an else.

I changed the `value` to be optional, so it can be used in that way.
If you don't give a value, the template is always rendered (besides if initializing and reconciling are given) and if you give a value and it does not match either noting, or if given, the else template is rendered. 

I also described that a bit more in the docs.
This API allows all the modes I think.

```html
<div *stringFeatureFlag="'themeColor'; value: 'dark'; default: 'light'; else: stringFeatureElse; let value; let details = evaluationDetails">
  It was a match!
  The theme color is {{ value }} because of {{ details.reason }}
</div>
<ng-template #stringFeatureElse let-value let-details="evaluationDetails">
  It was no match!
  The theme color is {{ value }} because of {{ details.reason }}
</ng-template>
```

```html
<div *stringFeatureFlag="'themeColor'; default: 'light'; let value;">
  The theme color is {{ value }}.
</div>
```

Reply:
>I'm not also not 100% sure there's much value in differentiating between the Initializing and reconciling templates; in react, we just show the suspense in both cases and I suspect that's probably all people would need.

I was also wondering about it but I think the way it is here gives most flexibility.
We had cases were we did not want to see the reconciling but the loading, you can do that by simply omitting the reconciling template here. 
If you want to have it for both, you can simply set the same "loading template" to both parameters.

```
<div *booleanFeatureFlag="'flag'; default: false; initializing: loading; reconciling: loading">
  This is shown when the feature flag matches the specified user configuration.
</div>
<ng-template #loading>
  This is shown when the feature flag is initializing or reconciling.
</ng-template>
```

Reply:
> Definitely, but it was already in. Or do you mean it differently @toddbaert? I added some docs for that :)

Oh, I'm sorry! I missed that.